front-end/src/components/EditProfile/index.js

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import styled from 'styled-components';
+import SideBarNav from '../../shared/SideBarNav';
+import jwt_decode from 'jwt-decode'
+import * as DashboardApi from '../../api/DashboardApi';
+import FormInput from "../../shared/FormInput";
+import { Link, withRouter } from "react-router-dom";
+
+
+
+class EditProfile extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: '',
+      email: '',
+      phone: '',
+      twitter: '',
+      github_username:'',
+      github_info: null,
+      isLoaded: false,
+      hasAuthError: false,
+      hasServerError: false,
+      hasGithubProfileNotFoundError: false,
+      token: ''
+    };
+  }
+
+  getJWTTokenAndID(){
+    const token = localStorage.getItem('access_token');
+    const decodedToken = jwt_decode(token);
+    const userID = decodedToken.id;
+
+    this.setState({
+      id: userID,
+      token: token
+    })
+
+    return decodedToken;
+  }
+
+  componentDidMount() {
+    this.getJWTTokenAndID();
+
+    this.setState({hasAuthError: false, hasServerError: false}, async () => {
+      try {
+        const res = await DashboardApi.pullUserInfo(this.state.id, this.state.token);
+        if (res.status === 200) {
+          this.setState({
+            isLoaded: true,
+            id: res.data[0].id,
+            email: res.data[0].email,
+            twitter: res.data[0].twitter || '',
+            phone: res.data[0].phone || '',
+            github_info: res.data[0].github_info
+          });
+        }
+      } catch (e) {
+        const {response: {status}} = e
+        if (status >= 400 && status < 500) {
+          this.setState({hasAuthError: true});
+        } else {
+          this.setState({hasServerError: true});
+        }
+      }
+    });
+  }
+
+  handleChange = (field, event) => {
+    this.setState({
+      [field]: event.target.value
+    })
+  }
+
+
+  handleGithubProfileFormSubmit = async (event) =>{
+    event.preventDefault();
+
+    this.setState({hasAuthError: false, hasServerError: false}, async () => {
+      try {
+        const res = await DashboardApi.linkGithubAccount(this.state.id, this.state.token, this.state.github_username);
+        if (res.status === 200) {
+          this.props.history.push('/dashboard');
+        }
+      } catch (e) {
+
+        const {response: {status}} = e
+        if ( status === 404) {
+          this.setState({hasGithubProfileNotFoundError: true});
+        }
+        else if (status >= 405 && status < 500) {
+          this.setState({hasAuthError: true});
+        } else {
+          this.setState({hasServerError: true});
```

Comment:
At some point, I'd like to separate all of this logic into an `ApiClient` class that will call whatever endpoints and then handle any errors